### PR TITLE
3466 Fix "trim of undefined" on checkout login

### LINKED ISSUE #3466: Error occurred while trying to register upon checkout TypeError: Cannot read property 'trim' of undefined
### Expected behavior
When I want to register a new user and I don't input anything in the login form field, I should  errors stating empty value .

### Actual behavior
 When I don't enter anything in the form field I get an error below and an infinite spinner.

 ```
Uncaught TypeError: Cannot read property 'trim' of undefined
    at AuthContainer._this.handleFormSubmit (auth.js:43)
    at SignUp._this.handleSubmit (signUp.js:39)
    at HTMLUnknownElement.boundFunc (modules.js?hash=7ac780ae72799aa31b599a547745df7e0444d087:132276)
    at Object.ReactErrorUtils.js.ReactErrorUtils.invokeGuardedCallback (modules.js?hash=7ac780ae72799aa31b599a547745df7e0444d087:132282)
    at executeDispatch (modules.js?hash=7ac780ae72799aa31b599a547745df7e0444d087:132063)
    at Object.executeDispatchesInOrder (modules.js?hash=7ac780ae72799aa31b599a547745df7e0444d087:132086)
    at executeDispatchesAndRelease (modules.js?hash=7ac780ae72799aa31b599a547745df7e0444d087:131480)
    at executeDispatchesAndReleaseTopLevel (modules.js?hash=7ac780ae72799aa31b599a547745df7e0444d087:131491)
    at Array.forEach (<anonymous>)
    at forEachAccumulated (modules.js?hash=7ac780ae72799aa31b599a547745df7e0444d087:132385)
```
### Steps to reproduce the behavior
- Order a product without login or registering
- Add to cart
- Checkout
- Click on register without filling the form field
- Notice the error above

### Versions
```
Node: 8.9.1
NPM: 5.5.1
Meteor Node: 8.9.3
Meteor NPM: 5.5.1
Reaction CLI: 0.24.2
Reaction: 1.6.4
Reaction branch: master
```

### PR DESCRIPTION
**Pull request by iakhator**
resolve #3466 

This PR fixes trim of undefined when registering/login on checkout. I found out that the trim() method automatically trims an empty field. what I did was to check if the field is undefined then reset it to an empty string else trim it, then perform the necessary validations.

#### Test
- Order a product without login or registering
- Add to cart
- Checkout
- Click on register without filling the form field
- Notice the validation error display for empty field

### CODE REVIEW

File: imports/plugins/core/accounts/client/containers/auth.js [hunk 1]
```diff
@@ -40,8 +40,8 @@ class AuthContainer extends Component {
       isLoading: true
     });
     const errors = {};
-    const username = email.trim();
-    const pword = password.trim();
+    const username = email === undefined ? "" : email.trim();
+    const pword = password === undefined ? "" : password.trim();
```
Comment (pmn4) [inline imports/plugins/core/accounts/client/containers/auth.js]: should we consider `null` as well? if so, maybe use a truthy/falsey test instead? `email ? "" : email.trim()`

**Comment by pmn4** [comment]
lgtm

**Comment by iakhator** [comment]
@pmn4 alright, I will get it done.

File: imports/plugins/core/accounts/client/containers/auth.js (hunk 1 above)
Comment (zenweasel) [reply imports/plugins/core/accounts/client/containers/auth.js]: Testing for equality with `undefined` is a no-no as the results aren't deterministic. Use `typeof` instead.

**Pull request merged**